### PR TITLE
Added login with QR and Auth system changed from access token to key/secret

### PR DESCRIPTION
**Pull request by openwarrior**
Intent integration is done for calling Scanner application of ZXing.
Auth system is changed from access token to key secret pair, to remove dependency from keycloak.
Implemented AuthzModule is used instead of complete Aerogear authz to sustain connectivity with pipe.
Changes still needed to be commented in better way. Working on that

### CODE REVIEW

File: src/main/res/values/strings.xml [hunk 1]
```diff
@@ -34,7 +34,7 @@
 
   <string name="label_authorization_description">Please provide the server location. The next step will require the credentials.</string>
   <string name="label_authorization_title">Authorization using a Hawkular server is necessary.</string>
-  <string name="label_authorization_support">The supported Hawkular version is 1.0.0 Alpha 3.</string>
+  <string name="label_authorization_support">The supported Hawkular version is 1.0.0 Alpha 9.</string>
```
Comment (jpkrohling) [inline src/main/res/values/strings.xml]: I think you can change to Alpha10, as it depends on stuff that was not present on Alpha 9.

File: src/main/java/org/hawkular/client/android/util/ModifiedAuthzModule.java [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+                    HeaderAndBody result;
+                    try {
+                        HttpProvider provider =
+                                new HttpRestProvider(new URL(((LoginActivity) (activity)).backendUrl.toString()
+                                        + "/secret-store/v1/tokens/create"));
+                        provider.setDefaultHeader("Accept", "application/json");
+                        provider.setDefaultHeader("Authorization", "Basic "
+                                + buildLoginData(((LoginActivity) (activity)).username,
+                                ((LoginActivity) (activity)).password));
+                        result = provider.post("a");
```
Comment (jpkrohling) [inline src/main/java/org/hawkular/client/android/util/ModifiedAuthzModule.java]: "a"? You don't need to post anything to this endpoint. It seems you can just pass `null` to `HttpRestProvider#post()`

File: src/main/java/org/hawkular/client/android/util/ModifiedAuthzModule.java [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+                        Log.d("gotta", new String(provider.getUrl().toString()));
+                        Log.d("gotta", new String(result.getBody()));
```
Comment (jpkrohling) [inline src/main/java/org/hawkular/client/android/util/ModifiedAuthzModule.java]: Better logging: 

```
Log.d("Called with POST to URL: ", new String(provider.getUrl().toString()));
Log.d("Result of the POST: ", new String(result.getBody()));
```

File: src/main/java/org/hawkular/client/android/util/ModifiedAuthzModule.java [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+        return false;
+    }
+
+    @Override
+    public void requestAccess(final Activity activity, final Callback<String> callback) {
+
+        Exception exception;
+        String json;
+        if (Preferences.of(activity).exist().get().equals("true")) {
```
Comment (jpkrohling) [inline src/main/java/org/hawkular/client/android/util/ModifiedAuthzModule.java]: Is this guaranteed to not cause a `NullPointerException`?

File: src/main/java/org/hawkular/client/android/util/ModifiedAuthzModule.java (hunk 2 above)
Comment (openwarrior) [inline src/main/java/org/hawkular/client/android/util/ModifiedAuthzModule.java]: need string but doesn't care what.
so should I send empty string then

File: src/main/java/org/hawkular/client/android/util/ModifiedAuthzModule.java (hunk 3 above)
Comment (openwarrior) [inline src/main/java/org/hawkular/client/android/util/ModifiedAuthzModule.java]: done

File: src/main/java/org/hawkular/client/android/util/ModifiedAuthzModule.java (hunk 2 above)
Comment (jpkrohling) [inline src/main/java/org/hawkular/client/android/util/ModifiedAuthzModule.java]: You sure it has to be an String? Wouldn't `null` suffice ?

File: src/main/java/org/hawkular/client/android/util/ModifiedAuthzModule.java (hunk 4 above)
Comment (openwarrior) [inline src/main/java/org/hawkular/client/android/util/ModifiedAuthzModule.java]: Took code from cookbook of aerogear
Not sure

File: src/main/java/org/hawkular/client/android/util/ModifiedAuthzModule.java (hunk 2 above)
Comment (openwarrior) [inline src/main/java/org/hawkular/client/android/util/ModifiedAuthzModule.java]: giving error with null

Comment (pilhuhn) [inline src/main/java/org/hawkular/client/android/util/ModifiedAuthzModule.java]: @danielpassos Want to chime in?

**Comment by openwarrior** [comment]
finally :dancers:  :D

File: build.gradle [hunk 5]
```diff
@@ -161,4 +161,6 @@ dependencies {
     androidTestCompile "com.android.support.test:rules:0.3"
     androidTestCompile "com.android.support.test:runner:0.3"
     androidTestCompile "com.squareup.assertj:assertj-android:1.1.0"
+
+    compile 'com.google.zxing:android-integration:3.2.1'
```
Comment (arturdryomov) [inline build.gradle]: Please follow the common formatting style. Dependencies are listed alphabetically, grouped by Gradle specifier. And double quotes of course.

File: src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -63,20 +63,31 @@
       android:label="@string/title_metrics">
     </activity>
 
+
```
Comment (arturdryomov) [inline src/main/AndroidManifest.xml]: Please remove unnecessary empty lines and use common indentation for XML.

File: src/main/java/org/hawkular/client/android/activity/AuthorizationActivity.java [hunk 7]
```diff
@@ -16,40 +16,29 @@
...
-
 import butterknife.Bind;
 import butterknife.ButterKnife;
 import butterknife.OnClick;
-import timber.log.Timber;
 
 /**
  * Authorization activity.
- *
+ * <p>
```
Comment (arturdryomov) [inline src/main/java/org/hawkular/client/android/activity/AuthorizationActivity.java]: What is the reason for this?

File: src/main/java/org/hawkular/client/android/activity/AuthorizationActivity.java [hunk 8]
```diff
@@ -98,7 +87,11 @@ public void setUpAuthorization() {
             return;
         }
 
-        setUpBackendAuthorization();
+        Intent login_activity = new Intent(getApplicationContext(), LoginActivity.class);
```
Comment (arturdryomov) [inline src/main/java/org/hawkular/client/android/activity/AuthorizationActivity.java]: There is the `Intents` factory-like class specifically for this. And please use camel case for variables declaration.

Comment (ming13) [inline src/main/java/org/hawkular/client/android/activity/AuthorizationActivity.java]: There is the `Intents` factory-like class specifically for this. And please use camel case for variables declaration.

File: src/main/java/org/hawkular/client/android/activity/LoginActivity.java [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+
+
+    private void setUpBindings() {
+        ButterKnife.bind(this);
+    }
+
+    private void setUpToolbar() {
+        setSupportActionBar(toolbar);
+    }
+
```
Comment (arturdryomov) [inline src/main/java/org/hawkular/client/android/activity/LoginActivity.java]: Too many empty lines.

File: src/main/java/org/hawkular/client/android/backend/BackendClient.java [hunk 10]
```diff
@@ -66,11 +66,7 @@
  * Most of the configuration is stored using internal AeroGear long-lived objects. It is not necessary to handle
  * this class objects as singletons, it is intended to be used as a short-lived object.
  *
- * As a rule of thumb, it is necessary to configure authorization using {@link #configureAuthorization(String)}
```
Comment (arturdryomov) [inline src/main/java/org/hawkular/client/android/backend/BackendClient.java:69]: Why was this removed?

File: src/main/java/org/hawkular/client/android/util/ModifiedAuthzModule.java [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+
+import android.app.Activity;
+import android.content.Context;
+import android.os.AsyncTask;
+import android.util.Base64;
+import android.util.Log;
+
+public class ModifiedAuthzModule implements AuthzModule {
+
+    Context applicationContext;
```
Comment (arturdryomov) [inline src/main/java/org/hawkular/client/android/util/ModifiedAuthzModule.java]: I’m pretty sure it is not a good idea to use package-scoped fields when it can be even `final`.

File: src/main/res/values-w820dp/dimens.xml [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<resources>
+  <!-- Example customization of dimensions originally defined in res/values/dimens.xml
```
Comment (arturdryomov) [inline src/main/res/values-w820dp/dimens.xml]: Placeholder.

File: src/main/res/values/dimens.xml [hunk 13]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <!--
 
-    Copyright 2015 Red Hat, Inc. and/or its affiliates
+    Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
```
Comment (arturdryomov) [inline src/main/res/values/dimens.xml]: The file wasn’t modified, why year change?

**Comment by arturdryomov** [comment]
@anuj1708 Please try to follow the common style used in the application. I’m not really nitpicking (yeah, I’m pretty sure about that), but it is really hard to judge your changes when more than a half of them are unnecessary code reformattings and inconsistencies.

The second important note — please try to describe your change at least briefly. Take a look at #28 for example. It allows people like me understand what are you actually trying to achieve. The current title contains grammatical errors and doesn’t really say much about your proposed changes.

File: build.gradle (hunk 5 above)
Comment (openwarrior) [inline build.gradle]: Thanks.
As you can see I am new to open source :)
will learn gradually

File: src/main/AndroidManifest.xml (hunk 6 above)
Comment (openwarrior) [inline src/main/AndroidManifest.xml]: just used the formatter provided by hawkular and copy pasted one to add only one added activity.
So indentation is that is used by hawkular only

File: src/main/java/org/hawkular/client/android/activity/AuthorizationActivity.java (hunk 7 above)
Comment (openwarrior) [inline src/main/java/org/hawkular/client/android/activity/AuthorizationActivity.java]: Came up by mistake from next line

File: src/main/java/org/hawkular/client/android/activity/AuthorizationActivity.java (hunk 8 above)
Comment (openwarrior) [inline src/main/java/org/hawkular/client/android/activity/AuthorizationActivity.java]: Sir you might have not noticed but here Authz is not being used and is continued only coz of sustaining next functionalities. From discussion with summers and suggestion of jpkrohling. 
Authz cant provide feature of using key/secret

File: src/main/java/org/hawkular/client/android/backend/BackendClient.java (hunk 10 above)
Comment (openwarrior) [inline src/main/java/org/hawkular/client/android/backend/BackendClient.java:69]: I removed that method so comment was creating a error

File: src/main/res/values/dimens.xml (hunk 13 above)
Comment (openwarrior) [inline src/main/res/values/dimens.xml]: Mistake in the fuss.
command modified it somehow and i didn't noticed.
Added by mistake with rest of commits

File: src/main/java/org/hawkular/client/android/util/ModifiedAuthzModule.java (hunk 11 above)
Comment (openwarrior) [inline src/main/java/org/hawkular/client/android/util/ModifiedAuthzModule.java]: I will remember that :)

File: src/main/res/values-w820dp/dimens.xml (hunk 12 above)
Comment (openwarrior) [inline src/main/res/values-w820dp/dimens.xml]: didn't  got you

**Comment by openwarrior** [comment]
@ming13 Yeah. It been a bit of mess but clearing it out.
Focus of my code was using QR scanning. Given by @jpkrohling 
Making independent of Keycloak. Given by jpkrohling
And for both of them I had to change authentication system.

And It really contain some real work too. Even though I understand I myself was feeling by being new to Git I have made a lot of things look bad as changed code and formating.

But ones the formatting was changed. I could not do anything but to correct that back and which somehow looked like changing of code.

I respect you sir and your code too. You may check you have several of my mails for help :)

**Comment by danielpassos** [comment]
@anuj1708 Sorry to not review that before. It was a holiday in my country :). Are you about to send this again?

**Comment by openwarrior** [comment]
Yeah. But that will take me some time :smile:
